src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -270,10 +271,12 @@ protected static RelationshipTermModel relationshipTermModelFromColumn(Table tab
   public static AssetSpecification assetModelToAssetSpecification(
       AssetModel assetModel,
       Map<String, DatasetTable> tables,
-      Map<String, Relationship> relationships) {
+      Map<String, Relationship> datasetRelationships) {
     AssetSpecification spec = new AssetSpecification().name(assetModel.getName());
+    List<String> assetRelationships =
+        Objects.requireNonNullElse(assetModel.getFollow(), new ArrayList<>());
```

Comment:
Since `processAssetRelationships()` doesn't modify its arguments you can use an immutable empty list here.
```suggestion
        Objects.requireNonNullElse(assetModel.getFollow(), List.of());
```